### PR TITLE
APIGOV-22093 Update GRPC Handlers and tests

### PR DESCRIPTION
**Pull request by tsjohns9**
- Set status to error on resources when processing fails
- Update tests to handle deleting state

### CODE REVIEW

File: pkg/agent/handler/accessrequest.go [hunk 1]
```diff
@@ -88,41 +88,50 @@ func (h *accessRequestHandler) onPending(ar *mv1.AccessRequest) error {
...
+		return
 	}
+
 	status := h.prov.AccessRequestDeprovision(req)
 
 	ri, _ := ar.AsInstance()
 	if status.GetStatus() == prov.Success {
 		h.client.UpdateResourceFinalizer(ri, arFinalizer, "", false)
+	} else {
+		h.onError(ar, fmt.Errorf(status.GetMessage()))
```
Comment (tsjohns9) [inline pkg/agent/handler/accessrequest.go:127]: Set status to error when deprovision failed

File: pkg/util/subresource.go [hunk 2]
```diff
@@ -61,6 +62,11 @@ func GetAgentDetailsValue(h handler, key string) (string, error) {
 // Returns errors if unable to perform type conversion.
 // Returns an empty string if the value does not exist, or if there is an error.
 func GetSubResourcePropertyValue(h handler, subRes, key string) (string, error) {
+	// check for a nil value, or a pointer to a nil value
+	if h == nil || reflect.ValueOf(h).Kind() == reflect.Ptr && reflect.ValueOf(h).IsNil() {
```
Comment (tsjohns9) [inline pkg/util/subresource.go:66]: Checking for nil, or a nil pointer to avoid panics

File: pkg/util/util.go [hunk 3]
```diff
@@ -248,3 +250,12 @@ func ToString(v interface{}) string {
 	}
 	return s
 }
+
+// AssertError asserts on the existence of an error for tests
+func AssertError(t *testing.T, hasErr bool, err error) {
```
Comment (jcollins-axway) [inline pkg/util/util.go:255]: I don't like pulling in the testing and assert packages for releases, that means all agents will have this package as well. Could we revert the use of this function?

**Review by jcollins-axway — CHANGES_REQUESTED** [review]

File: pkg/util/subresource.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/util/subresource.go:66]: maybe this check is worth moving to util function?

Comment (tsjohns9) [reply pkg/util/subresource.go:66]: I like that idea

File: pkg/util/util.go (hunk 3 above)
Comment (tsjohns9) [reply pkg/util/util.go:255]: Yeah

File: pkg/util/subresource.go [hunk 4]
```diff
@@ -63,7 +62,7 @@ func GetAgentDetailsValue(h handler, key string) (string, error) {
 // Returns an empty string if the value does not exist, or if there is an error.
 func GetSubResourcePropertyValue(h handler, subRes, key string) (string, error) {
 	// check for a nil value, or a pointer to a nil value
-	if h == nil || reflect.ValueOf(h).Kind() == reflect.Ptr && reflect.ValueOf(h).IsNil() {
+	if isNil := IsNil(h); isNil {
```
Comment (jcollins-axway) [inline pkg/util/subresource.go:65]: if the func returns a bool, no need for this var.

How are you pulling this in from the util package?

Comment (jcollins-axway) [reply pkg/util/subresource.go:65]: This is the util package, haha

**Review by jcollins-axway — APPROVED** [review]

File: pkg/util/subresource.go (hunk 4 above)
Comment (tsjohns9) [reply pkg/util/subresource.go:65]: haha :)
I'll remove the var

**Pull request merged**